model-extension-example/src/main/java/uk/gov/gchq/magmacore/examples/extensions/ext/HqdmExtensionService.java

```diff
@@ -16,21 +19,29 @@ public class HqdmExtensionService implements ExtensionService {
     /**
      * {@inheritDoc}
      */
+    @SuppressWarnings("unchecked")
```

Comment:
It just removes the warning about the unsafe cast to type `T`, which is raised because you can't check whether an object is of type `T` or not at runtime. The fact that the method returns `T` also means that the calling code doesn't need an ugly cast as well, but it does leave the door open for errors such as:
```java
final Child child = service.createEntity(`parent`, iri);
```
which will create a `Parent` and try to cast it to a `Child` entity which results in a `ClassCastException`. We can just make the factory return `Thing` and let the calling code do the type checking and casting if you prefer, but we do something similar in the `MagmaCoreService` I believe.